servers/src/main/java/tachyon/master/block/BlockMaster.java

```diff
@@ -0,0 +1,571 @@
+        if (workerInfo != null) {
+          locations.add(new BlockLocation(masterBlockLocation.getWorkerId(),
+              workerInfo.getAddress(), masterBlockLocation.getTier()));
+        }
+      }
+      return new BlockInfo(masterBlockInfo.getBlockId(), masterBlockInfo.getLength(), locations);
+    }
+    throw new BlockInfoException("Block info not found for " + blockId);
+  }
+
+  /**
+   * Retrieves information for the given list of block ids.
+   *
+   * @param blockIds A list of block ids to retrieve the information for
+   * @return A list of {@link BlockInfo} objects corresponding to the input list of block ids. The
+   *         list is in the same order as the input list.
+   */
+  public List<BlockInfo> getBlockInfoList(List<Long> blockIds) {
+    List<BlockInfo> ret = new ArrayList<BlockInfo>(blockIds.size());
+    for (long blockId : blockIds) {
+      MasterBlockInfo masterBlockInfo = mBlocks.get(blockId);
+      if (masterBlockInfo != null) {
+        // Construct the block info object to return.
+
+        // "Join" to get all the addresses of the workers.
+        List<BlockLocation> locations = new ArrayList<BlockLocation>();
+        for (MasterBlockLocation masterBlockLocation : masterBlockInfo.getBlockLocations()) {
+          MasterWorkerInfo workerInfo =
+              mWorkers.getFirstByField(mIdIndex, masterBlockLocation.getWorkerId());
+          if (workerInfo != null) {
+            locations.add(new BlockLocation(masterBlockLocation.getWorkerId(),
+                workerInfo.getAddress(), masterBlockLocation.getTier()));
+          }
+        }
+        BlockInfo retInfo =
+            new BlockInfo(masterBlockInfo.getBlockId(), masterBlockInfo.getLength(), locations);
+        ret.add(retInfo);
+      }
+    }
+    return ret;
+  }
+
+  /**
+   * @return the total bytes on each storage tier.
+   */
+  public List<Long> getTotalBytesOnTiers() {
+    List<Long> ret = new ArrayList<Long>(Collections.nCopies(StorageLevelAlias.SIZE, 0L));
+    synchronized (mWorkers) {
+      for (MasterWorkerInfo worker : mWorkers) {
+        for (int i = 0; i < worker.getTotalBytesOnTiers().size(); i ++) {
+          ret.set(i, ret.get(i) + worker.getTotalBytesOnTiers().get(i));
+        }
+      }
+    }
+    return ret;
+  }
+
+  /**
+   * @return the used bytes on each storage tier.
+   */
+  public List<Long> getUsedBytesOnTiers() {
+    List<Long> ret = new ArrayList<Long>(Collections.nCopies(StorageLevelAlias.SIZE, 0L));
+    synchronized (mWorkers) {
+      for (MasterWorkerInfo worker : mWorkers) {
+        for (int i = 0; i < worker.getUsedBytesOnTiers().size(); i ++) {
+          ret.set(i, ret.get(i) + worker.getUsedBytesOnTiers().get(i));
+        }
+      }
+    }
+    return ret;
+  }
+
+  public long getWorkerId(NetAddress workerNetAddress) {
+    // TODO: this NetAddress cloned in case thrift re-uses the object. Does thrift re-use it?
```

Comment:
Sure, this was done before, so I transferred it to here. But yes, we need some sort of test to verify this.